design/authenticators/authn_oidc/oidc_configuration_enhancements.md

```diff
@@ -0,0 +1,228 @@
+OpenIDConnect.debug!
+# Include cert-related debug messages?
+OpenSSL.debug = true
+```
+Running the above in an initializer and will enable debug logging on the OpenIDConnect gem and all its 
+relevant dependencies and ensure that they get routed to the Rails logger. We could also include the debug 
+logs for OpenSSL if we need to specifically focus on certificate-related debugging, which seems to be a 
+common sticking point for authn-oidc users.
+
+### Components
+1. OpenIDConnect - primary gem for [generating an OIDC client](https://github.com/cyberark/conjur/tree/master/app/domain/authentication/authn_oidc/v2/client.rb#L106C20-L106C20) in 
+authn-oidc which can retrieve authorization codes and tokens from the OIDC provider. It outsources discovery 
+to a couple other small gems which we need to consider as well:
+1. SWD (Simple Web Discovery) - handles OIDC configuration discovery against the provider URI configuration 
+endpoint: `/.well-known/openid-configuration` 
+1. WebFinger - used for dynamic discovery when the credential issuer is unknown to the client. **Not used in Conjur.**
+
+### Considerations around OIDC Discovery
+The SWD gem discussed previously is used by OpenIDConnect across many authenticators in Conjur (authn-oidc, 
+authn-jwt, authn-azure, authn-gcp). It performs a couple common tasks in Conjur by invoking the OIDC discovery 
+endpoint:
+  - validate authenticator status and provider connectivity
+  - fetch provider keys for decoding tokens
+
+This should not interfere with the proposed solution, but it is important to note since the modified code will be 
+used across different authenticators to invoke the provider discovery endpoint `/.well-known/openid-configuration`. 
+It will be necessary that the other authenticators are unaffected by this change.
+
+### Testing
+The testing will depend on the final implementation, but at the very least the wrapper method will need to be 
+well-covered by unit tests.
+
+#### Unit tests (temporary truststore modifications)
+1. OpenSSL truststore contains the CA cert symlink while yielding
+1. OpenSSL truststore symlink is removed after yielding
+1. Hash collision in truststore does not produce an error but instead increments the file extension of the symlink
+1. Temp file/dir exists while yielding
+1. Temp file/dir contains expected certificate content while yielding
+1. Temp file/dir is removed after yielding
+
+#### Unit tests (authenticator data object)
+1. Default `ca-cert` value is nil
+1. Set value matches expected
+
+#### Cucumber tests
+One or more of our Keycloak OIDC scenarios should be updated to use a `ca-cert` value provided via Conjur 
+variable. It may turn out to be easier to update all keycloak examples to use the cert variable, and the case 
+where the variable is not set can be covered by the Okta examples.
+
+### Open Questions
+- Do we need validation that `ca-cert` is a valid cert before attempting to use it?
```

Comment:
Do we have a sense of how much additional work it will be to add the validation? If its not a valid cert, whats the worst thats going to happen?

Reply:
In all likelihood this will be a non-issue now that I've thought about it more. We will use the OpenSSL Ruby library to parse the cert(s) from the `ca-cert` value, and it will produce an error we can easily handle if it is malformed in any way. 

The worst that will happen is the TLS connection would fail and they would have to fix the variable value.

Reply:
We also need to explore what might happen with malicious certs, not just malformed.

Reply:
@gl-johnson @jvanderhoof 

love the collaboration in this PR 👍  - I was wondering if the contents of the `ca-certs` variable (ie: cert or cert-chain) represent the "trust-store" only for the oidc provider or proxy and would this impact other calls or session refresh?  What if you had multiple providers (okta & identity)?  Additionally, if you have a customer cert and a proxy would proxy CA cert include the cert for the oidc provider?  Also, Could we leverage the system/os trust store/cert store instead of using an variable?  Also, are we trimming spaces and CRLF?